job/handlers/resources/WindowsServer_2016/gitRepo/templates/providers/_bitbucket.ps1

```diff
@@ -71,19 +71,40 @@ $SUBSCRIPTION_PRIVATE_KEY_PATH = @'
 <%=subPrivateKeyPath%>
 '@
 
-Function git_sync() {
+Function add_project_key() {
   $ssh_dir = Join-Path "$global:HOME" ".ssh"
   $key_file_path = Join-Path "$ssh_dir" "id_rsa"
 
   if (Test-Path $key_file_path) {
-    echo "----> Removing $key_file_path"
+    echo "----> Removing key: $key_file_path"
     Remove-Item -Force $key_file_path
   }
   [IO.File]::WriteAllLines($key_file_path, $PRIVATE_KEY)
   & $env:OPENSSH_FIX_USER_FILEPERMS
 
+  echo "----> Adding project key to ssh agent"
+  exec_exe "ssh-agent"
+  exec_exe "ssh-add $key_file_path"
+}
+
+Function add_subscription_key() {
+  $ssh_dir = Join-Path "$global:HOME" ".ssh"
+  $key_file_path = Join-Path "$ssh_dir" "id_rsa"
```

Comment:
Why are we replacing id_rsa here? On other platforms we do `00_project` and `01_sub`. Won't that work on Windows, too?

Reply:
Windows SSH agent has a limitation that, it could only add one key at a time ( which is at path ~/.ssh/id_rsa ), that's why we added copying contents to id_rsa and then adding it to ssh in our first iteration of support for windows git repos cloning.

Reply:
Wow. Not much of an agent, then :)

Is that documented somewhere?

Reply:
We have come across this already in https://github.com/Shippable/pm/issues/10002

To be accurate, the problem is not exactly with ssh-agent itself, but the script that is expected to run for fixing permissions ( [documented here at Win32-OpenSSH](https://github.com/PowerShell/Win32-OpenSSH/wiki/OpenSSH-utility-scripts-to-fix-file-permissions) ), which we run before doing ssh-add.

```
& $env:OPENSSH_FIX_USER_FILEPERMS 
exec_exe "ssh-agent"
exec_exe "ssh-add $key_file_path"
```

`& $env:OPENSSH_FIX_USER_FILEPERMS ` script fixes permissions only for key at `~/.ssh/id_rsa`.

Reply:
Ok, this means things like submodules will not work at all. I wasn't aware of this limitation. It might make sense for us to fork the script and make it do what we need.

Not blocking this merge for it, but we need to investigate.